### PR TITLE
[FXML-4414] Implement leftover variants of cmpf

### PR DESCRIPTION
**Pull request by cferry-AMD**

### CODE REVIEW

**Review by TinaAMD — APPROVED** [review]
LGTM

File: mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp [hunk 1]
```diff
@@ -79,6 +80,13 @@ Value createCheckIsOrdered(ConversionPatternRewriter &rewriter, Location loc,
                                               firstIsNaN, secondIsNaN);
 }
 
+emitc::ConstantOp getConstant(OpBuilder rewriter, Location loc,
+                              llvm::APInt val) {
+  auto type = rewriter.getIntegerType(val.getBitWidth());
+  return rewriter.create<emitc::ConstantOp>(loc, type,
+                                            rewriter.getIntegerAttr(type, val));
```
Comment (mgehre-amd) [inline mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp:87]:
```suggestion
                                            rewriter.getBoolAttr(type, val));
```
might be enough here for creating boolean constants

Comment (josel-amd) [reply mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp:87]: Sure, I can make this specific to i1

**Review by josel-amd — APPROVED** [review]

**Comment by cferry-AMD** [comment]
@josel-amd , I let you merge this when you think it's ready :)

**Pull request merged**